icu4c/source/test/intltest/rbbitst.cpp

```diff
@@ -4098,53 +4287,72 @@ void RBBITest::RunMonkey(BreakIterator *bi, RBBIMonkeyKind &mk, const char *name
                     }
                 }
 
-                // Format looks like   "<data>\\\uabcd\uabcd\\\U0001abcd...</data>"
-                UnicodeString errorText = "<data>";
-                /***if (strcmp(errorType, "next()") == 0) {
-                    startContext = 0;
-                    endContext = testText.length();
+                // Formatting of each line includes:
+                //   character code
+                //   reference break: '|' -> a break, '.' -> no break
+                //   actual break:    '|' -> a break, '.' -> no break
+                //   (name of character clase)
+                //   Unicode name of character
+                //   '-->' indicates location of the difference.
 
-                    printStringBreaks(testText, expected, expectedCount);
-                }***/
+                UErrorCode status = U_ZERO_ERROR;
+                const char *badLocale = bi->getLocaleID(ULOC_ACTUAL_LOCALE, status);
```

Comment:
The failing locale will always be root - it's the only thing the old monkey test handles, so can be omitted here.

One additional thing from the new monkey test that has proven useful are the command line test invocation parameters for reproducing the test case. Copy and Paste ready. Saves having to dig up the somewhat obscure syntax for passing test parameters (the seed and break iterator type) into the monkeytest.

Reply:
Done

Reply:
Updated the message to include command line parameters.